packages/compass-preferences-model/src/global-config.ts

```diff
@@ -0,0 +1,277 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import { EJSON } from 'bson';
+import yaml from 'js-yaml';
+import type { Options as YargsOptions } from 'yargs-parser';
+import yargsParser from 'yargs-parser';
+import type { AmpersandType, GlobalPreferences } from './preferences';
+import { allPreferencesProps } from './preferences';
+
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+const { log, mongoLogId } = createLoggerAndTelemetry('COMPASS-PREFERENCES');
+
+function getGlobalConfigPaths(): string[] {
+  const paths = [];
+
+  if (process.env.COMPASS_GLOBAL_CONFIG_FILE_FOR_TESTING) {
+    paths.push(process.env.COMPASS_GLOBAL_CONFIG_FILE_FOR_TESTING);
+  }
+
+  switch (process.platform) {
+    case 'win32':
+      if (process.execPath === process.argv[1]) {
+        paths.push(path.resolve(process.execPath, '..', 'mongodb-compass.cfg'));
+      }
+      break;
+    default:
+      paths.push('/etc/mongodb-compass.conf');
+      break;
+  }
+  return paths;
+}
+
+async function loadGlobalPreferences(
+  globalConfigPaths: string[]
+): Promise<[config: unknown, file: string]> {
+  let file = '';
+  let filename = '';
+  let config: unknown;
+
+  for (filename of globalConfigPaths) {
+    try {
+      file = await fs.readFile(filename, 'utf8');
+      break;
+    } catch (error: any) {
+      log[error?.code === 'ENOENT' ? 'info' : 'warn'](
+        mongoLogId(1_001_000_153),
+        'preferences',
+        'Skipping global configuration file due to error',
+        {
+          filename,
+          error: error?.message,
+          code: error?.code,
+        }
+      );
+    }
+  }
+
+  if (!file) {
+    return [{}, '<no global configuration file>'];
+  }
+
+  log.info(
+    mongoLogId(1_001_000_154),
+    'preferences',
+    'Loading global configuration file',
+    { filename }
+  );
+
+  try {
+    if (file.trim().startsWith('{')) {
+      config = EJSON.parse(file);
+    } else {
+      config = yaml.load(file);
+    }
+  } catch (error: any) {
+    log.error(
+      mongoLogId(1_001_000_155),
+      'preferences',
+      'Invalid global configuration file',
+      {
+        filename,
+        error: error?.message,
+      }
+    );
+  }
+
+  return [config, filename];
+}
+
+const cliProps = Object.entries(allPreferencesProps).filter(
+  ([, definition]) => definition.cli
+);
+function getCliPropNamesByType(type: AmpersandType<any>): string[] {
+  return cliProps
+    .filter(([, definition]) => definition.type === type)
+    .map(([key]) => key);
+}
+
+const yargsOptions: YargsOptions = {
+  string: [...getCliPropNamesByType('string')],
+  boolean: [...getCliPropNamesByType('boolean')],
+  number: [...getCliPropNamesByType('number')],
```

Comment:
Super tiny nit, doesn't look like we need to spread here?

```suggestion
  string: getCliPropNamesByType('string'),
  boolean: getCliPropNamesByType('boolean'),
  number: getCliPropNamesByType('number'),
```

Reply:
Thank you, yes :slightly_smiling_face: (Originally I had put some options like `--help` not as entries in the preferences description list, but only added them to the parser here, so it was something like `[...getCliPropNamesByType('boolean'), 'help'], then decided the special handling would introduce more complexity then it saves, removed them again, and forgot to remove the array spreading here :slightly_smiling_face:)

Removed in 57e7ff75dfaf2ba2268fd651f41eaef16d9ba858!